### PR TITLE
feat:(1058) - Implemented the design change of Search SPA

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 1058

# Explain the feature
Implemented the Search SPA design

1. Redesigned the search ui
2. Updated the filter views and sort
2. Included the Not found page in search

## Does this PR introduce a breaking change
Yes

## Screenshots

<details>
<summary>View Screenshots</summary>

1) Not found Page
localhost_5500_search__query=DSAL

2) Search results page
localhost_5500_search__query=feedback

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/search-spa/set-env.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const targetPath = './src/environments/environment.ts';
 require('dotenv').config();
 
 const envConfigFile = `export const environment = {
-   API_URL: process.env.API_URL,
+   API_URL: '${process.env.API_URL}',
```
Comment (deshmukhmayur) [inline packages/search-spa/set-env.ts:10]: Use a fallback when using interpolation in javascript templates for potentially undefined values.
```suggestion
   API_URL: '${process.env.API_URL || ''}',
```

Comment (deshmukhmayur) [reply packages/search-spa/set-env.ts:10]: The above line can cause problems down the line if API_URL is undefined.
```ts
console.log(typeof a)
// > undefined

console.log(typeof `${a}`)
// > string
```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]

**Review by deshmukhmayur — APPROVED** [review]

File: packages/search-spa/set-env.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ const targetPath = './src/environments/environment.ts';
 require('dotenv').config();
 
 const envConfigFile = `export const environment = {
-   API_URL: process.env.API_URL,
+   API_URL: '${process.env.API_URL || ''}',
```
Comment (ghanlohar) [inline packages/search-spa/set-env.ts:10]:
```suggestion
   API_URL: process.env.API_URL || '',
```

**Review by ghanlohar — CHANGES_REQUESTED** [review]

File: packages/search-spa/set-env.ts (hunk 2 above)
Comment (deshmukhmayur) [reply packages/search-spa/set-env.ts:10]: @ghanlohar It's already in a string template, so this won't work.

Comment (riginoommen) [reply packages/search-spa/set-env.ts:10]: This actually breaks the env file generation.

Comment (ghanlohar) [reply packages/search-spa/set-env.ts:10]: Yeah I see. Looks good. Thanks.

**Review by ghanlohar — APPROVED** [review]

**Pull request merged**